refactor(migrations): initial migration logic for converting to signal queries

devversion opened the pull request:
Adds initial migration logic to convert decorator query declarations to signal queries.

We will re-use more part of the signal input migration in follow-ups, to properly migrate,
and e.g. even handle control flow

devversion committed:
refactor(migrations): initial migration logic for converting to signal queries

Adds initial migration logic to convert decorator query declarations to
signal queries.

We will re-use more part of the signal input migration in follow-ups, to
properly migrate, and e.g. even handle control flow

crisbeto commented on packages/core/schematics/migrations/signal-queries-migration/convert_query_property.ts:
```diff
@@ -0,0 +1,133 @@
+  const defaultDescendants = metadata.kind !== 'contentChildren';
+  const optionProperties: ts.PropertyAssignment[] = [];
+  const args: ts.Expression[] = [
+    metadata.args[0], // Locator.
```
Isn't it possible for this to be undefined?

devversion replied:
Probably not. I will add some safety fallback to `undefined` then.

devversion replied:
Seems like it's guaranteed to exist given this is metadata parsed by the Angular compiler.

devversion committed:
refactor(migrations): properly handle multi query migration

Properly handles queries with multiple results, by extracting the
type from the `QueryList`.

Also adds more tests and handles imports.

pkozlowski-opensource commented on packages/core/schematics/migrations/signal-queries-migration/identify_queries.ts:
```diff
@@ -0,0 +1,119 @@
+import {R3QueryMetadata} from '../../../../compiler';
+
+/** Branded type to uniquely identify class properties in a project. */
+export type ClassPropertyID = UniqueID<'ClassPropertyID; Potentially an output'>;
```
Potentially an output => Potentially a query?

devversion replied:
good catch. Fixed

devversion committed:
fixup! refactor(migrations): initial migration logic for converting to signal queries

Fix incorrect ID description

alxhub commented:
This PR was merged into the repository by commit f454ad3bcff801895d8fa1e4316bdb5da8b7368c.

The changes were merged into the following branches: main, 18.2.x
